src/Middleware/SpaServices.Extensions/src/DevelopmentServer/DevelopmentServerMiddleware.cs

```diff
@@ -54,43 +56,42 @@ internal static class ReactDevelopmentServerMiddleware
                 // the first request times out, subsequent requests could still work.
                 var timeout = spaBuilder.Options.StartupTimeout;
                 return targetUriTask.WithTimeout(timeout,
-                    $"The create-react-app server did not start listening for requests " +
+                    $"The {serverName} server did not start listening for requests " +
                     $"within the timeout period of {timeout.Seconds} seconds. " +
                     $"Check the log output for error information.");
             });
         }
 
-        private static async Task<int> StartCreateReactAppServerAsync(
-            string sourcePath, string npmScriptName, ILogger logger)
+        private static async Task<int> StartDevServerAsync(
+            string sourcePath, string npmScriptName, string waitText, string serverName, ILogger logger, IDictionary<string, string> extraArgs)
         {
             var portNumber = TcpPortFinder.FindAvailablePort();
-            logger.LogInformation($"Starting create-react-app server on port {portNumber}...");
+            logger.LogInformation($"Starting {serverName} server on port {portNumber}...");
 
             var envVars = new Dictionary<string, string>
             {
-                { "PORT", portNumber.ToString() },
-                { "BROWSER", "none" }, // We don't want create-react-app to open its own extra browser window pointing to the internal dev server port
+                { "PORT", portNumber.ToString() }
             };
             var npmScriptRunner = new NpmScriptRunner(
-                sourcePath, npmScriptName, null, envVars);
+                sourcePath, npmScriptName, null, envVars.Union(extraArgs));
```

Comment:
I'm aware this will cause a NullReference. I'll setup a local build env and possibly add some tests.